tests/test_cleanup_pods.py

```diff
@@ -80,8 +85,27 @@ def test_cleanup_api_exception_continue(load_incluster_config, list_namespaced_p
     delete_pod.side_effect = kubernetes.client.rest.ApiException(status=0)
     pod1 = MagicMock()
     pod1.metadata.name = 'dummy'
+    pod1.metadata.labels = {'airflow_kpo_in_cluster': False}
     pod1.status.phase = 'Succeeded'
     pod1.status.reason = None
     list_namespaced_pod().items = [pod1]
     cleanup('awesome-namespace')
     load_incluster_config.assert_called_once()
+
+
+@mock.patch('astronomer.cleanup_pods.command_line.delete_pod')
+@mock.patch('kubernetes.client.CoreV1Api.list_namespaced_pod')
+@mock.patch('kubernetes.config.load_incluster_config')
+def test_cleanup_kpo_pods_delay(load_incluster_config, list_namespaced_pod, delete_pod):
+    pod1 = MagicMock()
+    pod1.metadata.name = 'dummy'
+    pod1.metadata.labels = {'airflow_kpo_in_cluster': True}
+    pod1.status.phase = 'Succeeded'
+    pod1.status.reason = None
+    container1 = MagicMock()
+    container1.last_transition_time = datetime.now(timezone.utc) - timedelta(seconds=3599)
```

Comment:
Let's leave a little more buffer here.
```suggestion
    container1.last_transition_time = datetime.now(timezone.utc) - timedelta(seconds=300)
```